### PR TITLE
adjusts to SpaWebFilter.

### PR DESCRIPTION
**Pull request by mshima**
Related to https://github.com/jhipster/generator-jhipster/pull/20107#issuecomment-1414109437

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/java/package/web/filter/SpaWebFilter.java.ejs [hunk 1]
```diff
@@ -35,14 +35,19 @@ public class SpaWebFilter extends OncePerRequestFilter {
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
         throws ServletException, IOException {
         String path = request.getRequestURI();
-        if (!path.startsWith("/api") && !path.startsWith("/management")<% if (authenticationTypeOauth2) { %> && !path.startsWith("/login") && !path.startsWith("/oauth2")<% } %>
-<%_ if (applicationTypeGateway || applicationTypeMonolith) { _%>
-            && !path.startsWith("/services") && !path.startsWith("/v3/api-docs")
+        if (
+            !path.startsWith("/api")
+            && !path.startsWith("/management")
+            && !path.startsWith("/v3/api-docs")
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/web/filter/SpaWebFilter.java.ejs:41]: `/v3/api-docs` exists in microservices.

File: generators/server/templates/src/main/java/package/web/filter/SpaWebFilter.java.ejs [hunk 2]
```diff
@@ -35,14 +35,19 @@ public class SpaWebFilter extends OncePerRequestFilter {
...
+<%_ if (authenticationTypeOauth2 && applicationTypeMonolith) { _%>
+            && !path.startsWith("/login")
+            && !path.startsWith("/oauth2")
 <%_ } _%>
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/web/filter/SpaWebFilter.java.ejs:45]: Oauth2 urls should not be exposed at microservices.

File: generators/server/templates/src/main/java/package/web/filter/SpaWebFilter_reactive.java.ejs [hunk 3]
```diff
@@ -31,14 +31,22 @@ public class SpaWebFilter implements WebFilter {
...
+        if (
+            !path.startsWith("/api")
+            && !path.startsWith("/management")
+            && !path.startsWith("/v3/api-docs")
+<%_ if (authenticationTypeOauth2 && (applicationTypeGateway || applicationTypeMonolith)) { _%>
+            && !path.startsWith("/login")
+            && !path.startsWith("/oauth2")
+<%_ } _%>
+<%_ if (applicationTypeGateway) { _%>
+            && !path.startsWith("/services")
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/web/filter/SpaWebFilter_reactive.java.ejs:43]: `/services` only exists at gateways

File: generators/server/templates/src/main/java/package/web/filter/SpaWebFilter.java.ejs (hunk 1 above)
Comment (mraible) [reply generators/server/templates/src/main/java/package/web/filter/SpaWebFilter.java.ejs:41]: When I create a new project, Prettier formats this like the following:

```java
@Override
protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
    throws ServletException, IOException {
    String path = request.getRequestURI();
    if (
        !path.startsWith("/api") &&
        !path.startsWith("/management") &&
        !path.startsWith("/v3/api-docs") &&
        !path.contains(".") &&
        path.matches("/(.*)")
    ) {
        request.getRequestDispatcher("/index.html").forward(request, response);
        return;
    }

    filterChain.doFilter(request, response);
}
```

**Review by mraible — APPROVED** [review]

**Pull request merged**